sklearn/tests/test_kernel_approximation.py

```diff
@@ -91,7 +97,12 @@ def test_polynomial_count_sketch_dense_sparse(gamma, degree, coef0, csr_containe
 
 
 def _linear_kernel(X, Y):
-    return np.dot(X, Y.T)
+    if X.ndim == 1 and Y.ndim == 1:
+        return X @ Y
+    elif X.ndim == 2 and Y.ndim == 2:
+        return X @ Y.T
```

Comment:
codecov is complaining about this case not being covered, but i don't think the `ndim == 2 thing` will ever happen because of how we defined [_pairwise_callable](https://github.com/scikit-learn/scikit-learn/blob/396edeba21339554ec32e8d2e591fbbec900651a/sklearn/metrics/pairwise.py#L1991)... maybe we should just remove the conditional check altogether?